KarhooSDK/Api/Util/Result.swift

```diff
@@ -9,23 +9,33 @@
 import Foundation
 
 public enum Result<T> {
-    case success(result: T)
-    case failure(error: KarhooError?)
+    case success(result: T, correlationId: String? = nil)
+    case failure(error: KarhooError?, correlationId: String? = nil)
```

Comment:
I'd like to empathize how much I dislike the need to do such hacks. We need to do it, but I'd like to make it clear :D